CapstoneProject/Assets/Scripts/Core/Controller/Dialogue/DialogueManager.cs

```diff
@@ -87,37 +129,97 @@ public bool LoadScriptObject()
     {
         bool success = true;
 
+        //Loads the dialogue based on the current scene NPC
+        if (gameState.currentNPC == "Whirl")
+        {
+            MoveObjectToList(whirlDialogue);
+            displayPortraitObject.GetComponent<UnityEngine.UI.Image>().sprite = whirlPortraitImage;
+        }
+        else if (gameState.currentNPC == "Emelia")
+        {
+            MoveObjectToList(emeliaDialogue);
+            displayPortraitObject.GetComponent<UnityEngine.UI.Image>().sprite = emeliaPortraitImage;
+        }
+
+
+        //Check if the dialogue loaded successfully
+        if (loadedDialogue == null)
+        {
+            UnityEngine.Debug.Log("Error: Failed to load dialogue.");
+            success = false;
+        }
+        if (displayPortraitObject.GetComponent<UnityEngine.UI.Image>().sprite == null)
+        {
+            UnityEngine.Debug.Log("Error: Failed to load dialogue portrait.");
+            success = false;
+        }
+        
 
         return success;
     }
 
+    public void MoveObjectToList(DialogueExample dialogueObject)
+    {
+        //The first dialogue element is always the transition line.
+        dialogueTransition = dialogueObject.dialogueText[0];
+
+        //Moves the dialogue from the script to the list
+        for (int i = 0; i < dialogueObject.dialogueText.Count - 1; i++)
+        {
+            loadedDialogue.Add(dialogueObject.dialogueText[i + 1]);
+            //Debug.Log(loadedDialogue[i]);
+        }
+
+        //Prints out contents of both lists for debugging
+        //dialogueObject.dialogueText.ForEach(UnityEngine.Debug.Log);
+        //loadedDialogue.ForEach(UnityEngine.Debug.Log);
+    }
+
     public void StartInteraction()
     {
         DisplayDialogueBox();
         DisplayDialoguePortrait();
         BeginText();
     }
 
+    //Enables both UI elements
     private void DisplayDialogueBox()
     {
-
+        displayBoxObject.GetComponent<UnityEngine.UI.Image>().enabled = true;
     }
 
     private void DisplayDialoguePortrait()
     {
-
+        displayPortraitObject.GetComponent<UnityEngine.UI.Image>().enabled = true;
     }
 
     private void BeginText()
     {
+        displayTextObject.SetActive(true);
+        displayTextObject.GetComponent<TextMeshProUGUI>().enabled = true;
+        StartCoroutine(TextScroll("This is test dialogue. Wa ba ba go bo"));
 
+        
+    }
+
+    IEnumerator TextScroll(string displayText)
+    {
+        for (int i = 0; i < displayText.Length; i++)
+        {
+            displayTextObject.GetComponent<TextMeshProUGUI>().SetText(displayText.Substring(0, i+1));
+            yield return new WaitForSeconds(textScrollSpeed);
```

Comment:
There's a new allocation happening each letter here, perhaps do a single new outside the loop and then use it here.

Reply:
Clarification: set WaitForSeconds outside of for loop, call in for loop